File: apps/api/app/schemas/escrow.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class EscrowEventRead(BaseModel):
	model_config = ConfigDict(from_attributes=True)

	id: str
	job_id: str
	milestone_id: str | None
	event_type: str
	stellar_tx_hash: str | None
	payload: dict | None
	created_at: datetime


class EscrowStatusRead(BaseModel):
	escrow_id: str
	contract_address: str
	job_status: str
	on_chain_status: str | None = None
	last_synced_at: datetime | None = None

File: packages/types/src/escrow.ts
import type { JobStatus } from "./job";
import type { MilestoneStatus } from "./milestone";

export type EscrowEventType = "FUNDED" | "APPROVED" | "DISPUTED" | "RESOLVED" | "RELEASED";

export interface EscrowEvent {
	id: string;
	jobId: string;
	milestoneId: string | null;
	eventType: EscrowEventType;
	stellarTxHash: string | null;
	payload: Record<string, unknown> | null;
	createdAt: string;
}

export interface EscrowMilestoneSnapshot {
	sequence: number;
	status: MilestoneStatus;
	releasedAmount: string;
}

export interface EscrowSnapshot {
	jobId: string;
	escrowId: string;
	contractAddress: string;
	status: JobStatus;
	totalReleasedAmount: string;
	milestones: EscrowMilestoneSnapshot[];
	lastSyncedAt: string | null;
}
